ui/app/AppLayouts/Chat/controls/community/HoldingsDropdown.qml

```diff
@@ -38,275 +39,263 @@ StatusDropdown {
-                StatusSwitchTabButton {
-                    text: qsTr("Token")
-                    fontPixelSize: 13
-                }
-                StatusSwitchTabButton {
-                    text: qsTr("Collectible")
-                    fontPixelSize: 13
-                }
-                StatusSwitchTabButton {
-                    text: qsTr("ENS")
-                    fontPixelSize: 13
-                    enabled: false // TODO
+
+            tabLabels: [qsTr("Token"), qsTr("Collectible"), qsTr("ENS")]
+            state: [d.tokensState, d.collectiblesState, d.ensState][currentIndex]
+
+            currentIndex: d.currentTabIndex
+            onCurrentIndexChanged: d.currentTabIndex = currentIndex
+
+            onAddClicked: d.addClicked()
+
+            Connections {
+                target: backButton
+
+                function onClicked() {
+                    content.state = d.operatorsState
                 }
             }
-            Loader {
-                id: tabsLoader
-                Layout.fillWidth: true
-                Layout.margins: 8
-            }
         }
     }
 
     Component {
-        id: tokensCollectiblesLayout
+        id: tokensLayout
 
-        ColumnLayout {
-            spacing: 0
-            StatusPickerButton {
-                Layout.fillWidth: true
-                Layout.preferredHeight: 36
-                bgColor: Theme.palette.baseColor5
-                contentColor: Theme.palette.directColor1
-                text: d.isTokensLayout ? root.tokenName : root.collectibleName
-                font.pixelSize: 13
-                image.source: d.isTokensLayout ? root.tokenImage : root.collectibleImage
-                onClicked: loader.sourceComponent = extendedView
+        TokensPanel {
+            id: tokensPanel
+
+            tokenName: root.tokenName
+            tokenImage: root.tokenImage
+            amount: root.tokenAmount === 0 ? "" : root.tokenAmount.toString()
+            onAmountChanged: root.tokenAmount = Number(amount)
+
+            onPickerClicked: {
+                d.extendedDropdownType = ExtendedDropdownContent.Type.Tokens
+                content.state = d.extendedState
             }
 
-            RowLayout {
-                visible: !d.isTokensLayout
-                Layout.fillWidth: true
-                Layout.alignment: Qt.AlignVCenter
-                Layout.leftMargin: 16
-                Layout.rightMargin: 6
-                Layout.topMargin: 8
-                StatusBaseText {
-                    Layout.fillWidth: true
-                    text: qsTr("Specific amount")
-                    font.pixelSize: 13
-                    wrapMode: Text.WordWrap
-                    elide: Text.ElideRight
-                    clip: true
+            Connections {
+                target: d
+
+                function onAddClicked() {
+                    root.addItem(root.itemKey, root.tokenAmount.toString() + " " + root.tokenName,
```

Comment:
`root.tokenAmount.toString() + " " + root.tokenName`

Nope :) This needs to
- format the number
- use `arg()` for the order

Reply:
Basically, I think that formatting requires more attention and I would leave it as it is in this PR. It's not my code and I didn't refactored it in this PR. But right, it should be done but I would opt for doing it in a separate pr.

Some problems I see:
- currently FloatValidator is used, which is incompatible with localized formatting
- probably exponential notation should be avoided for token amount (when used "%L1 %2")
- this string should not be composed in this component, it was done this way just to have first version of gui and should be refactored
- there is a problem with singular/plural forms of token names

Reply:
I agree, this needs a complex and a higher level approach